great_expectations/data_context/data_context.py

```diff
@@ -3264,6 +3270,41 @@ def run_checkpoint(
             **kwargs,
         )
 
+    def add_profiler(
+        self,
+        name: str,
+        config_version: float,
+        rules: Dict[str, dict],
+        class_name: str,
+        module_name: Optional[str] = None,
+        variables: Optional[dict] = None,
+        commented_map: Optional[CommentedMap] = None,
+        ge_cloud_id: Optional[str] = None,
+    ):
+        config_data = {
```

Comment:
@cdkini Mind adding the type hint (for aesthetics purposes, of course).  Thanks! :-)

Reply:
I don't think we need type annotations for objects that are obvious; for example, `config_data` is very clearly a dictionary here.

I think type annotations make more sense in function signatures and places where there's variability but always adding them no matter what detracts from the readability of the code (in my opinion of course).

We should reflect the decision here in the style guide. cc @donaldheppner

Reply:
in this case the object's lifecycle is short lived and the IDE can fully infer the type. I'm fine omitting it in this case